src/librustc/hir/map/mod.rs

```diff
@@ -351,7 +352,7 @@ impl<'hir> Map<'hir> {
             Node::Field(_) => DefKind::Field,
             Node::Expr(expr) => match expr.kind {
                 ExprKind::Closure { .. } => DefKind::Closure,
-                _ => return None,
+                _ => bug!("local def id of expression"),
```

Comment:
`bug!("def_kind: unsupported node: {}", self.node_to_string(hir_id))`.

Same at the other `return None,`.